pkg/kubelet/client/kubelet_client.go

```diff
@@ -93,7 +93,8 @@ func (c *KubeletClientConfig) transportConfig() *transport.Config {
 			KeyFile:  c.KeyFile,
 			KeyData:  c.KeyData,
 		},
-		BearerToken: c.BearerToken,
+		BearerToken:       c.BearerToken,
+		ConnectionTimeout: 7 * time.Second,
```

Comment:
the timeouts for controller manager observing a node's status, and for a node to heartbeat with the apiserver are all tunable. if this timeout is intended to coordinate with those, it should be tunable as well